planer-uwr-webapp/imports/ui/pages/PlanPage/RulesetSummary.tsx

```diff
@@ -39,36 +46,57 @@ export const RulesetSummary = ({ ruleset, planId }: RulesetSummaryProps) => {
     return null;
   }
 
+  const is_passing = (rule: BaseRule) => filterCourses(courses, rule) >= (rule.condition === true ? 1 : rule.condition);
+  const score_ratio = (rule: BaseRule) => `${filterCourses(courses, rule)} / ${(rule.condition === true ? '1' : rule.condition)}`;
+
   return (
-    <div style={{ border: '1px solid deeppink', gridColumn: '1 / span 1000' }}>
-      <div>
-        <div>{ruleset.name}</div>
-        {ruleset.rules.map((rule, idx) => (
+    <div className='rule-row'>
+      <div className='title'>{ruleset.name}</div>
+      {ruleset.rules.map((rule, idx) => {
+        // Get corresponding brightness variant with gievn color palette
+        const rowBrightness = (colors: string[]) => (idx % 2 === 0)
+          ? `${colors[idx % 2]}, #344653`
+          : `${colors[idx % 2]}, #202B34`;
+        // Get corresponding grandient background
+        const rowBackground = is_passing(rule)
+          ? `linear-gradient(90deg, ${rowBrightness(bgColors['green'])} 10%, transparent 70%)`
+          : `linear-gradient(90deg, ${rowBrightness(bgColors['red'])} 10%, transparent 70%)`;
+        // Get corresponding text color
+        const rowColor = is_passing(rule)
+          ? 'rgb(103, 214, 103)'
+          : 'rgb(255, 74, 74)';
+        return (
           <div
             key={ruleset._id + rule.name + idx}
-            style={{ border: '1px solid maroon', padding: 4 }}
+            className='rule'
+            style={{ background: rowBackground }}
           >
-            <b>{rule.name}</b> {filterCourses(courses, rule)}{' '}
-            {filterCourses(courses, rule) >=
-            (rule.condition === true ? 1 : rule.condition)
-              ? 'GOOD'
-              : 'BAD'}
-            {rule.subRules &&
-              rule.subRules.map((rule, idx) => (
-                <span
-                  key={ruleset._id + rule.name + idx}
-                  style={{ border: '1px solid maroon', padding: 4 }}
-                >
-                  <b>{rule.name}</b> {/* {filterCourses(courses, rule)} */}{' '}
-                  {filterCourses(courses, rule) >=
-                  (rule.condition === true ? 1 : rule.condition)
-                    ? 'GOOD'
-                    : 'BAD'}
-                </span>
-              ))}
+            <div className='score' style={{ color: rowColor }}>
```

Comment:
Could we move that stuff info CSS classes? And utilize features like `:nth-child(even)`.